torchmetrics/functional/text/rouge.py

```diff
@@ -0,0 +1,156 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import re
+from typing import Dict, List, Tuple, Union
+
+import torch
+from torch import Tensor, tensor
+
+from torchmetrics.utilities.imports import _NLTK_AVAILABLE, _ROUGE_SCORE_AVAILABLE
+
+if _ROUGE_SCORE_AVAILABLE:
+    from rouge_score.rouge_scorer import RougeScorer
+    from rouge_score.scoring import AggregateScore, BootstrapAggregator
+else:
+    RougeScorer, AggregateScore, BootstrapAggregator = object, object, object
+
+ALLOWED_ROUGE_KEYS = (
+    "rouge1", "rouge2", "rouge3", "rouge4", "rouge5", "rouge6", "rouge7", "rouge8", "rouge9", "rougeL", "rougeLsum"
+)
+
+
+def add_newline_to_end_of_each_sentence(x: str) -> str:
+    """This was added to get rougeLsum scores matching published rougeL scores for BART and PEGASUS."""
+    if _NLTK_AVAILABLE:
+        import nltk
+        nltk.download("punkt", quiet=True, force=False)
+
+    re.sub("<n>", "", x)  # remove pegasus newline char
+    assert nltk, "nltk must be installed to separate newlines between sentences. (pip install nltk)"
+    return "\n".join(nltk.sent_tokenize(x))
+
+
+def format_rouge_results(result: Dict[str, AggregateScore], decimal_places: int = 4) -> Dict[str, Tensor]:
+    """Formats the computed (aggregated) rouge score to a dictionary of tensors format. """
+    flattened_result = {}
+    for rouge_key, rouge_aggregate_score in result.items():
+        for stat in ["precision", "recall", "fmeasure"]:
+            mid = rouge_aggregate_score.mid
+            score = round(getattr(mid, stat), decimal_places)
+            flattened_result[f"{rouge_key}_{stat}"] = tensor(score, dtype=torch.float)
+    return flattened_result
+
+
+def _rouge_score_update(
+    preds: List[str],
+    targets: List[str],
+    scorer: RougeScorer,
+    aggregator: BootstrapAggregator,
+    newline_sep: bool = False,
+) -> None:
```

Comment:
add docstring + example

Reply:
@Borda This is the update method used by the class and functional implementation. I have added docstring, Should I add an example as well ?

Reply:
well some minimal which would serve as unittest :]